File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        return dfs(numbers, target, 0, 0);
    }
    
    private int dfs(int[] numbers, int target, int i, int sum) {
        if (i == numbers.length) {
            return sum == target ? 1 : 0;
        }
        int usePlus  = dfs(numbers, target, i + 1, sum + numbers[i]);
        int useMinus = dfs(numbers, target, i + 1, sum - numbers[i]);
        return usePlus + useMinus;
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js

function dfs(numbers, target,  i, sum) {
        if (i == numbers.length) {
            return sum == target ? 1 : 0;
        }
        const usePlus  = dfs(numbers, target, i + 1, sum + numbers[i]);
        const useMinus = dfs(numbers, target, i + 1, sum - numbers[i]);
        return usePlus + useMinus;
}
    

function solution(numbers, target) {
    return dfs(numbers, target, 0, 0)
}
